ci/util.py

```diff
@@ -30,6 +31,45 @@ class Failure(RuntimeError, ValueError):
     pass
 
 
+class LintingError(Failure):
+    pass
+
+
+class LintingResult:
+    def __init__(self, problems):
+        problems_dict = collections.defaultdict(list)
+        for p in problems:
+            problems_dict[self._normalise_problem_level(p)].append(p)
+        self.problems = problems_dict
+
+    def _normalise_problem_level(self, p):
+        # Yamllint uses a dict with mappings for both int->level and level->int. Consistently use
+        # the int representation
+        if isinstance(p.level, str):
+            return yamllint.linter.PROBLEM_LEVELS[p.level]
+        else:
+            return p.level
+
+    def problems(self, level:int):
+        return self.problems_dict.get(level, ())
+
+    def has_problems(self):
+        return bool(self.problems)
+
+    def max_level(self):
+        if self.problems:
+            return sorted(self.problems.keys())[-1]
+        return 0
+
+    def problems_iter(self):
```

Comment:
tbh, I think this is a bit heavyweight, since it basically exposes the (almost same) functionality of a plain `dict.items()`